src/components/ListItem.jsx

```diff
@@ -1,5 +1,66 @@
 import './ListItem.css';
+import { updateItem } from '../api/firebase.js';
+import { useState, useEffect, useCallback } from 'react';
+import { ONE_DAY_IN_MILLISECONDS } from '../utils/dates.js';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ listPath, item }) {
+	const purchasedOneDayAgo = useCallback(() => {
+		if (item.dateLastPurchased !== null) {
+			const timeDiff = Date.now() - item.dateLastPurchased.seconds * 1000;
+			if (timeDiff <= ONE_DAY_IN_MILLISECONDS) {
+				return true;
+			} else {
+				return false;
+			}
+		} else {
+			return false;
+		}
+	}, [item.dateLastPurchased]);
+
+	const [isPurchased, setIsPurchased] = useState(false);
+	const [isChecked, setIsChecked] = useState(purchasedOneDayAgo);
+	function changeHandler(e) {
+		setIsPurchased(!isPurchased);
+		setIsChecked(!isChecked);
+	}
+
+	useEffect(() => {
+		async function purchaseItem() {
+			if (isPurchased) {
+				try {
+					await updateItem(listPath, item.id);
+				} catch (error) {
+					alert(error.message);
+				}
+			}
+		}
+		purchaseItem();
+	}, [isPurchased, item.id, listPath]);
+
+	useEffect(() => {
+		const timer = setTimeout(() => {
+			if (purchasedOneDayAgo) {
+				console.log(item.name);
+				setIsChecked(false);
+			}
+		}, 10000);
```

Comment:
I think this was set just for testing purposes but it will eventually need to be set to the difference between a day and the time difference of when it was last purchased. If you want to test this logic, you can just update the field in the Firebase store to be a day before you purchased

```suggestion
		}, ONE_DAY_IN_MILLISECONDS - (Date.now() - item.dateLastPurchased.seconds * 1000));
```